### PR TITLE
Add Tax ID type enum

### LINKED ISSUE #451: [Enhancement]: Add TIN Format Enum
### Description of Enhancement :

Enumerate some common TIN

### Why is this Needed?

Current Tax ID is unstructured string. At least trying to capture what format it represents would improve compatibility.

### PR DESCRIPTION
**Pull request by Sumukh**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it: 

Adds an Enum for Tax ID types

#### Which issue(s) this PR fixes:

Fixes #451

### CODE REVIEW

**Comment by Sumukh** [comment]
Please squash this before merging.

File: schema/enums/TaxIdentifierType.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "https://raw.githubusercontent.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/main/schema/enums/TaxIdentifierType.schema.json",
+  "title": "Enum - Tax Identifier Type",
+  "description": "Enumeration of tax ID types",
+  "type": "string",
+  "enum": ["SSN", "EIN", "ITIN", "PENDING", "OTHER"],
```
Comment (jacobyavis) [inline schema/enums/TaxIdentifierType.json:7]: This should probably be named as `UsTaxIdentifierType`, as the only options are applicable to United States.

We already have an issuing country code to help with tax identifier type (i.e. if the country code is CA, we assume it's a Canadian SIN), but what this seems to be doing is differentiating the many types of identifiers in use by the IRS

File: schema/enums/TaxIdentifierType.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "https://raw.githubusercontent.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/main/schema/enums/TaxIdentifierType.schema.json",
+  "title": "Enum - Tax Identifier Type",
+  "description": "Enumeration of tax ID types",
```
Comment (pjohnmeyer) [inline schema/enums/TaxIdentifierType.json:5]: Do we need additional information in the description here defining the values?

Comment (JSv4) [inline schema/enums/TaxIdentifierType.json:5]:
```suggestion
  "description": "The tax_id_type enumeration defines the category of a Tax Identifier Number that an individual or entity can possess. The options within this enum include 1). SSN (Social Security Number): A number assigned to U.S. citizens and some residents to track their earnings and benefits; 2). EIN (Employer Identification Number): A number that identifies a business entity in the U.S.; 3). ITIN (Individual Taxpayer Identification Number): A tax processing number issued to individuals in the U.S. who are required to have a taxpayer identification number but do not have or are not eligible to get an SSN; 4). PENDING: Indicates that a tax ID application is in process; and 5). OTHER: For any other types of tax IDs not listed above, typically used for identifiers issued by countries other than the U.S.",
```

File: schema/enums/TaxIdentifierType.json (hunk 1 above)
Comment (JSv4) [reply schema/enums/TaxIdentifierType.json:7]: Yeah, we wanted to add some structure to this so we could more reliably parse out common TAX IDs types and then do validation on the values. Do we want to have separate enums for different countries? Or do we want to use prefixes for now - e.g. US_SSN, US_EIN, CA_SIN, etc.?

**Comment by Sumukh** [comment]
Vote is to kill it.

**Comment by JSv4** [comment]
We decided this wasn't going go add a ton of value.

**Pull request closed without merging**